x/profiles/keeper/handler.go

```diff
@@ -250,3 +254,33 @@ func handleMsgAcceptDTagTransfer(ctx sdk.Context, keeper Keeper, msg types.MsgAc
 	return &result, nil
 
 }
+
+// DeleteDTagTransferRequest delete the request made by the sender and returns the event with the given eventType.
+func DeleteDTagTransferRequest(ctx sdk.Context, keeper Keeper, owner, sender sdk.AccAddress, eventType, deletionType string) (*sdk.Result, error) {
+	if err := keeper.DeleteDTagTransferRequest(ctx, owner, sender, deletionType); err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
+	}
+
+	ctx.EventManager().EmitEvent(sdk.NewEvent(
+		eventType,
+		sdk.NewAttribute(types.AttributeRequestReceiver, owner.String()),
+		sdk.NewAttribute(types.AttributeRequestSender, sender.String()),
+	))
+
+	result := sdk.Result{
+		Data:   keeper.Cdc.MustMarshalBinaryLengthPrefixed(sender),
+		Events: ctx.EventManager().Events(),
+	}
+
+	return &result, nil
+}
+
+// handleMsgRefuseDTagTransfer handles the reject of a dTag transfer request
+func handleMsgRefuseDTagTransfer(ctx sdk.Context, keeper Keeper, msg types.MsgRefuseDTagTransferRequest) (*sdk.Result, error) {
+	return DeleteDTagTransferRequest(ctx, keeper, msg.Sender, msg.Receiver, types.EventTypeDTagTransferRefuse, "refused")
+}
+
+// handleMsgCancelDTagTransfer handles the deletion of a dTag transfer request
+func handleMsgCancelDTagTransfer(ctx sdk.Context, keeper Keeper, msg types.MsgCancelDTagTransferRequest) (*sdk.Result, error) {
+	return DeleteDTagTransferRequest(ctx, keeper, msg.Receiver, msg.Sender, types.EventTypeDTagTransferCancel, "cancelled")
```

Comment:
Probably we should put `"cancelled"` as some sort of string inside the `types` package to make sure it's constant